### PR TITLE
Bug 1695228 - RLB: API to pass in raw samples for timing distributions

### PR DESCRIPTION
**Pull request by badboy**
Unfortunately this needs a new API instead of the already existing `accumulate_samples_signed`.
We want the RLB consumer (in this case: FOG) to not care about the unit,
so we expect nanosecond samples.
Internally we then do the right thing.

And because this is pure Rust we can rely on `u64` and just never need
to filter out negative samples.

### CODE REVIEW

File: glean-core/src/metrics/timing_distribution.rs [hunk 1]
```diff
@@ -314,6 +314,64 @@ impl TimingDistributionMetric {
...
+    ///
+    /// # Arguments
+    ///
+    /// * `samples` - A list of samples recorded by the metric.
+    ///               Samples must be in nanoseconds.
+    /// ## Notes
+    ///
+    /// Reports an [`ErrorType::InvalidOverflow`] error for samples that
+    /// are longer than `MAX_SAMPLE_TIME`.
+    /// Samples must be in nanoseconds, based on time from a monotonic clock.
```
Comment (chutten) [inline glean-core/src/metrics/timing_distribution.rs:327]: Do they really have to be based on a monotonic clock? I mean, I guess they have to be nonnegative (because they're `u64`) which requires them to be from an increasing clock? Not sure this phrase is adding much.

File: glean-core/src/metrics/timing_distribution.rs [hunk 2]
```diff
@@ -314,6 +314,64 @@ impl TimingDistributionMetric {
...
+    pub fn accumulate_raw_samples_nanos(&mut self, glean: &Glean, samples: &[u64]) {
+        if !self.should_record(glean) {
+            return;
+        }
+
+        let mut num_too_long_samples = 0;
+        let min_sample_time = self.time_unit.as_nanos(1);
```
Comment (chutten) [inline glean-core/src/metrics/timing_distribution.rs:334]: TIL zeros are forbidden

File: glean-core/tests/timing_distribution.rs [hunk 3]
```diff
@@ -334,3 +334,51 @@ fn stopping_non_existing_id_records_an_error() {
...
+    assert_eq!(1, snapshot.values[&2_784_941_737]);
+
+    // No errors should be reported.
+    assert!(test_get_num_recorded_errors(
+        &glean,
+        metric.meta(),
+        ErrorType::InvalidValue,
+        Some("store1")
+    )
+    .is_err());
```
Comment (chutten) [inline glean-core/tests/timing_distribution.rs:383]: Please also test minimum value and the error case.

**Review by chutten — APPROVED** [review]
Could land as-is, but I'd prefer if it came with one more test that checks two more things.

File: glean-core/src/metrics/timing_distribution.rs (hunk 1 above)
Comment (badboy) [reply glean-core/src/metrics/timing_distribution.rs:327]: They should, but I guess we take whatever as long as you promise it's some duration.

Comment (badboy) [reply glean-core/src/metrics/timing_distribution.rs:327]: (will change the sentence)

File: glean-core/src/metrics/timing_distribution.rs (hunk 2 above)
Comment (badboy) [reply glean-core/src/metrics/timing_distribution.rs:334]: Hey, I learned this as well.
